server/service/src/requisition/request_requisition/insert_program.rs

```diff
@@ -177,10 +194,46 @@ fn generate(
         .map(|line| line.item_id)
         .collect();
 
-    let requisition_line_rows =
+    let requisition_lines =
         generate_requisition_lines(ctx, &ctx.store_id, &requisition, program_item_ids)?;
 
-    Ok((requisition, requisition_line_rows))
+    let program_indicators = program_indicators(
+        connection,
+        Pagination::all(),
+        None,
+        Some(ProgramIndicatorFilter::new().program_id(EqualFilter::equal_to(&program.id))),
+    )?;
+
+    let supplier_store = StoreRepository::new(connection)
+        .query_one(StoreFilter::new().name_id(EqualFilter::equal_to(&other_party_id)))?;
+
+    // TODO Refactor check for store tags to less hard coded mechanism
+    let store_tags = NameTagRepository::new(&ctx.connection).query(Some(
+        NameTagFilter::new().store_id(EqualFilter::equal_to(&ctx.store_id)),
+    ))?;
+
+    let mut indicator_values: Vec<repository::IndicatorValueRow> = vec![];
+    let store_tag_names: Vec<String> = store_tags.into_iter().map(|t| t.name).collect();
+
+    if store_tag_names.contains(&ESPC_TAG.to_owned())
+        && !store_tag_names.contains(&NSPC_CLIENT_TAG.to_owned())
+    {
+        let indicators = match supplier_store {
+            Some(_) => {
+                generate_program_indicator_values(&ctx.store_id, &period_id, program_indicators)
+            }
+            None => vec![],
+        };
+        for indicator in indicators {
+            indicator_values.push(indicator)
+        }
```

Comment:
The generate program also generates a vec of indicators...

can you not indicator_values = ?